src/app/service/game-control.service.ts

```diff
@@ -49,155 +45,167 @@ export class GameControlService {
+        map(({ board, active, player }) => {
+          if (active && player === this.player) {
+            const arr = board;
+            arr[field] = player;
+            this.findWinner(arr, player);
+            updateDoc(doc(this.firestore, 'game', this.gameId), {
+              board: arr,
+              player: this.togglePlayer(player),
+            });
+          }
         }),
-        tap(() =>
-          this.activePlayer.next(
-            this.activePlayer.value === 'cross' ? 'circle' : 'cross'
-          )
-        ),
         take(1)
       )
       .subscribe();
   }
 
   reload() {
-    this.gameActive.next(true);
-    this.updateRound();
-    this.startWithCross = !this.startWithCross;
-    this.activePlayer.next(this.togglePlayer());
-    this.gameBoard.next(new Array(9).fill(Math.random(), 0, 9));
-    this.games$$.next(new ReplaySubject<move>());
+    this.gameData$
+      .pipe(
+        tap(({ round, startWith }) => {
+          updateDoc(doc(this.firestore, 'game', this.gameId), {
+            board: new Array(9).fill(null, 0, 9),
+            player: this.togglePlayer(startWith),
+            startWith: this.togglePlayer(startWith),
+            round: ++round,
+            active: true,
+          });
+        }),
+        take(1)
+      )
+      .subscribe();
   }
 
-  private findWinner(moves: move[], player: string) {
-    if (this.checkPattern(moves, player)) {
-      this.updateScore(player);
-    } else if (moves.length === 9) {
-      this.updateScore('');
+  deleteGame() {
+    if (confirm('Do you want to leave the game?')) {
+      deleteDoc(doc(this.firestore, 'game', this.gameId));
+      this.router.navigate(['']);
     }
   }
 
-  private getGameStatus(): Observable<boolean> {
-    return this.gameActive.asObservable();
-  }
-
-  private getScore(player: string): Observable<number> {
-    return this.score.asObservable().pipe(map((score) => score[player]));
-  }
-
-  private getWinner(): Observable<string> {
-    return this.winner.asObservable();
-  }
-
-  private filterPlayer(value: string, moves: move[]) {
+  private filterPlayer(value: string, moves: any[]) {
     return moves
-      .map(({ field, player }) => (player === value ? field + 1 : null))
+      .map((move, i) => (move === value ? i + 1 : null))
       .filter((el) => el);
   }
 
-  private updateScore(player: string) {
-    if (player) {
-      const score = { ...this.score.value };
-      ++score[player];
-      this.score.next(score);
-    }
-    this.winner.next(player);
-    this.gameActive.next(false);
+  private checkPattern(moves: any[], player: string) {
+    return this.patterns.some((pattern) =>
+      pattern.every((number) =>
+        this.filterPlayer(player, moves).includes(number)
+      )
+    );
   }
 
-  private updateRound() {
-    let round = this.round.value;
-    this.round.next(++round);
+  private findWinner(moves: any[], player: string) {
+    if (this.checkPattern(moves, player)) {
```

Comment:
I would also extract that game-checking logic into a separate service - so it would be easy to extend the rules easily. The game control service should work as an integration/coordination service - not an all-mighty service.

Reply:
done